src/main/java/org/broadinstitute/hellbender/tools/spark/sv/InsertSizeMetricsCollectorSpark.java

```diff
@@ -1,99 +1,256 @@
 package org.broadinstitute.hellbender.tools.spark.sv;
 
 import com.google.common.annotations.VisibleForTesting;
-import htsjdk.samtools.util.Histogram;
-import htsjdk.samtools.SAMReadGroupRecord;
 import htsjdk.samtools.SamPairUtil;
+import htsjdk.samtools.SAMFileHeader;
 import htsjdk.samtools.metrics.MetricsFile;
-import org.apache.spark.api.java.JavaRDD;
-import org.apache.commons.math3.stat.descriptive.DescriptiveStatistics;
-import org.broadinstitute.hellbender.tools.picard.analysis.InsertSizeMetrics;
+import htsjdk.samtools.util.Histogram;
 import org.broadinstitute.hellbender.utils.read.GATKRead;
+import org.broadinstitute.hellbender.metrics.MetricAccumulationLevel;
+import org.broadinstitute.hellbender.tools.picard.analysis.InsertSizeMetrics;
 
+import org.apache.spark.api.java.JavaRDD;
+
+import scala.Tuple2;
+
+import java.util.*;
+import java.util.function.Function;
+import java.util.stream.StreamSupport;
+import java.util.stream.Collectors;
 import java.io.Serializable;
-import java.util.List;
-import java.util.Map;
-import java.util.TreeMap;
-import java.util.SortedMap;
 
 /**
  * Worker class to collect insert size metrics.
+ * TODO: filter reads based on length value (if too large), and/or minimum_pct like in Picard.
+ * TODO: is it desired that mapping quality is collected as well?
+ * TODO: handle higher-case lower-case group name when asking for a particular group's stats
```

Comment:
This means that when someone asks for information regarding a particular group, the name could be a mix of higher-case lower-case letters. A more robust way is to convert all request to lowercase (or ignore case completely) to fetch information. Was just providing a comment to my self for next round improvements.